temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowRunLockManager.java

```diff
@@ -27,20 +27,11 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.ReentrantLock;
 
-final class WorkflowRunLockManager {
+public final class WorkflowRunLockManager {
   private final Map<String, RefCountedLock> runIdLock = new ConcurrentHashMap<>();
```

Comment:
I know not changed in this PR, but would change to type `ConcurrentMap` for documentation/expectation purposes. Confused me at first read.

In fact, after review, maybe the documented type for this should be `ConcurrentHashMap` or at least document the expectations around `compute`, because `Map.compute` isn't thread-safe obviously and `ConcurrentMap.compute` can run the closure multiple times (which is bad for us because we have a side-effecting closure there). But `ConcurrentHashMap.compute` seems to guarantee a single execution.